addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoDeviceDefinitions.java

```diff
@@ -0,0 +1,255 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.io.neeo.NeeoConstants;
+import org.openhab.io.neeo.NeeoUtil;
+import org.openhab.io.neeo.ServiceContext;
+import org.openhab.io.neeo.internal.models.NeeoCapabilityType;
+import org.openhab.io.neeo.internal.models.NeeoDevice;
+import org.openhab.io.neeo.internal.models.NeeoDeviceChannel;
+import org.openhab.io.neeo.internal.models.NeeoDeviceType;
+import org.openhab.io.neeo.internal.serialization.NeeoCapabilityTypeSerializer;
+import org.openhab.io.neeo.internal.serialization.NeeoDeviceChannelSerializer;
+import org.openhab.io.neeo.internal.serialization.NeeoDeviceSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+
+/**
+ * This class manages device definitions. Device definitions map openHAB things/channels to NEEO device/capabilities.
+ * All device definition changes are saved to a JSON file in the user data folder.
+ *
+ * @author Tim Roberts - initial contribution
+ */
+public class NeeoDeviceDefinitions implements AutoCloseable {
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoDeviceDefinitions.class);
+
+    /** The lock for accessing uidToDevice */
+    private final Lock lock = new ReentrantLock();
+
+    /** Cross reference between a ThingUID and a NeeoDevice */
+    private final Map<ThingUID, NeeoDevice> uidToDevice = new HashMap<ThingUID, NeeoDevice>();
+
+    /** The gson used to save/restore device definitions */
+    private final Gson gson;
+
+    /** The service context */
+    private final ServiceContext context;
+
+    /** The openHAB to NEEO converter */
+    private final OpenHabToDeviceConverter converter;
+
+    /** Whether the default is to expose all devices (as accessories) */
+    private final boolean exposeAll;
+
+    /** The file we store definitions in */
+    private final File file = new File(NeeoConstants.FILENAME_DEVICEDEFINITIONS);
+
+    /**
+     * Create the object based on the {@link ServiceContext} and will read the definitions from the {@link #file}
+     *
+     * @param context the non-null {@link ServiceContext}
+     */
+    public NeeoDeviceDefinitions(ServiceContext context) {
+        Objects.requireNonNull(context, "context cannot be null");
+
+        this.context = context;
+        this.converter = new OpenHabToDeviceConverter(context);
+
+        final GsonBuilder gsonBuilder = NeeoUtil.getGsonBuilder();
```

Comment:
You seem to be creating the GsonBuiler with registering at 3 different places? The other 2 are in util classes.

Reply:
Sorry for all the comments - but I didn't understand what you were meaning on this one...